### PR TITLE
[REVIEW] AWS S3 service

### PR DESCRIPTION
**Pull request by SudipKatwal**
## Description
* AWS service implementation for common use

* [Asana Link](https://app.asana.com/0/0/1202310707681151/f)

### CODE REVIEW

File: aws/configuration.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package aws
```
Comment (Swechhya) [inline aws/configuration.go:1]: Isn't this also specific to s3, can we move it inside s3 folder?

**Review by Swechhya — CHANGES_REQUESTED** [review]
You seem to be missing the following functions:
- UploadImageToS3
- DownloadFileToPath
- UploadFileFromPath

File: aws/s3/s3_service.go [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+	item := Item{
+		Bucket: u.Host,
+		Key:    u.Path,
+	}
+
+	return &item, nil
+}
+
+func CreateBucket(client *s3.Client, bucket string) (*string, error) {
```
Comment (varanasiaditya) [inline aws/s3/s3_service.go:57]: Is it better to pass the client every time?
or use composition and have method and client in a specific struct?

```
type PhilS3 struct {
	Client *s3.Client
}


func (ps3 PhilS3)CreateBucket(bucket string) (*string, error) {
	input := &s3.CreateBucketInput{
		Bucket: aws.String(bucket),
	}

	_, err := ps3.Client.CreateBucket(context.Background(), input)
	if err != nil {
		return nil, err
	}

	// Before returning the key, create an s3 key (URI)
	item := Item{
		Bucket: bucket,
	}
	s3URI := item.URI()

	return &s3URI, nil
} 


```

**Review by varanasiaditya — APPROVED** [review]

**Review by narup — APPROVED** [review]

File: pkg/s3/s3_service.go [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+	return &item, nil
+}
+
+func (ps3 *S3Client) CreateBucket(bucket string) (*string, error) {
+	input := &s3.CreateBucketInput{
+		Bucket: aws.String(bucket),
+	}
+
+	_, err := ps3.Client.CreateBucket(context.Background(), input)
```
Comment (abhyuday) [inline pkg/s3/s3_service.go:62]: send context from calling function?

**Review by abhyuday — APPROVED** [review]

File: pkg/s3/s3_service.go (hunk 3 above)
Comment (varanasiaditya) [reply pkg/s3/s3_service.go:62]: +1

Comment (SudipKatwal) [reply pkg/s3/s3_service.go:62]: Don't need to send context.Background() as parameter while calling the function. It simply works with context.Background() parameter within package. So, I sent context.Background() as default.

**Review by rachitamaharjan — APPROVED** [review]

**Review by chawocecil — APPROVED** [review]

**Pull request merged**